torch/_subclasses/fake_tensor.py

```diff
@@ -2100,6 +2138,30 @@ def wrap(e: T) -> Union[T, FakeTensor]:
 
         return tree_map(wrap, r)
 
+    def create_symbolic_nested_int(
+        self, *, hint: Optional[Union[int, torch.SymInt]] = None
```

Comment:
similar q, but what is the invariant around the type of hint if it is not None (can it really be either any int/symint?). Or is it guaranteed to be a nested_int, and we just don't have a better python type annotation for it

Reply:
I'm expecting the hint to always be a plain nested int. So the annotation should be a `Optional[torch.SymInt]` and this might just be copypasta.

Reply:
Actually this was related to https://github.com/pytorch/pytorch/pull/130292#discussion_r1702041479, and this should be an `Optional[int]`!